File: New Cycle Shop/angular/src/app/cycle.ts
export class Cycle {
    id: number | undefined;
    brand: string | undefined;
    stock: number | undefined;
    numBorrowed: number | undefined;

    get numAvailable(): number | undefined {
        return this.stock !== undefined && this.numBorrowed !== undefined ? this.stock - this.numBorrowed : undefined;
    }
}

File: Day 16/cycles-rest-api/rest/src/main/java/com/cycles/rest/entity/Cycle.java
package com.cycles.rest.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "cycles")
public class Cycle {
    
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    private String color;
    private String brand;
    private int quantity;
    private int price;
    private int numBorrowed = 0;
    private boolean addToCart;

    public int getNumAvailable() {
        return quantity - numBorrowed;
    }
}